### ISSUE TITLE
asyncio has memory leak on exceptions with 0.6.9

### ISSUE THREAD

**Issue by levesquejf** [Nuitka/Nuitka#851]
The bug described in #593 was resolved in 0.6.8 but is back in 0.6.9.1 and 0.6.9.2.

Objects are leaked when a coroutine raises an exception.

Version 0.6.8.4 is working fine.

Easy to reproduce with the same reproducer: https://github.com/levesquejf/nuitka-memleak-on-exceptions.

**Comment by kayhayen** [comment]
When I checked my working cases with the current code, and stabe code, I couldn't find anything that would indicate a regression in this domain.

**Comment by kayhayen** [comment]
My bad, I forgot to actually compile aiohttp and not just the main program, I reproduced this indeed.

**Comment by kayhayen** [comment]
This is a regression that I bisected to this unfortunately, so I have to assume, it's completely different issue, I will have to try and isolate which part has regressed there. Notice, I rebased master and develop, so far pushed to factory, because of bisect unfriendly commits leading up to 0.6.9 release, moving a fixup done late to earlier, which changes the commit ids.

```
commit d7f4e33098677ab70a4c4c6a93a1bd1dd777ca6a (refs/bisect/bad)
Author: Kay Hayen <kay.hayen@gmail.com>
Date:   Sun Jul 12 15:50:52 2020 +0200

    Optimization: Refactor constant code generation heavily
    
    * Constants are now externally processed and accessed via per-module constant indexes.
    
    * Added dedicated binary for merging constant values from all modules.
    
    * Make sure built-in dictionaries are sorted so their position can be used for reference.
    
    * Removed special internal module and made it normal with automatic parameter variable
      releases.
    
    * Built-in loopup helpers now use strings instead of premade strings. This allows to use
      them for constant creation too.
    
    * Global constants are no longer special and are created in the same way, referenced with
      defines that are indexes to global constants array.
    
    * Freezer and meta path based importer now share their data by using the same stream
      and are in the same files.
```

**Comment by kayhayen** [comment]
Luckily I was able to find it in this huge change relatively quickly, in an unrelated change to the complex call helper functions, where apparently optimization and cleanup was included in the commit for no good reason, plus is causing the regression. I am now very tempted to rebase again, although it wasn't necessary, since I just broke history the other day, it won't matter much anymore. But first I better find out, which part precisely caused it.

**Comment by kayhayen** [comment]
The fix is on factory now. Obviously it's an important thing and will be part of a hotfixed 0.6.9.6 soon.

**Comment by kayhayen** [comment]
Basically what happened was that as part of the constants change, internal helper functions were moved to a form where they became normal functions with a feature to automatically release variables, so we the try/finally constructs to release parameter variables was removed, and supposedly replaced with an automatic addition of all parameter variables as automatic releases on function exit. 

However, that bit wasn't done, the omission didn't trigger any test errors. It ought to have affected more code cases, I will try and produce one, but current test coverage did not. Maybe, in your example, the complex calls were only done in case of an exception, and that's why you saw them in such a way that suggested a relationship with exceptions. 

But like I said, I need to add these tests. I am so very surprised, none of the existing reference count tests found anything, but very happy to have bisect at hand, spotting the bug cause in your example, would have been tremendously tedious I guess.

However, from what it seems, this might have revealed bugs, so this seems to need more work.

**Comment by kayhayen** [comment]
The fix is part of 0.6.9.6 and I had no memory leak with it anymore on factory.
